docs/solidity-guides/decryption/decrypt_details.md

```diff
@@ -1,34 +1,30 @@
 # Decryption in depth
 
-This document provides a detailed guide on implementing decryption in your smart contracts using the `GatewayContract` in fhevm. It covers the setup, usage of the `Gateway.requestDecryption` function, and testing with Hardhat.
+This document provides a detailed guide on implementing decryption in your smart contracts using the `DecryptionOracle` in fhevm. It covers the setup, usage of the `FHE.requestDecryption` function, and testing with Hardhat.
 
-## `GatewayContract` set up
+## `DecryptionOracle` set up
 
-The `GatewayContract` is pre-deployed on the fhevm testnet. It uses a default relayer account specified in the `PRIVATE_KEY_GATEWAY_RELAYER` or `ADDRESS_GATEWAY_RELAYER` environment variable in the `.env` file.
+The `DecryptionOracle` is pre-deployed on the FHEVM testnet. It uses a default relayer account specified in the `PRIVATE_KEY_GATEWAY_RELAYER` or `ADDRESS_GATEWAY_RELAYER` environment variable in the `.env` file.
 
-Relayers are the only accounts authorized to fulfill decryption requests. The role of the `GatewayContract`, however, is to independently verify the KMS signature during execution. This ensures that the relayers cannot manipulate or send fraudulent decryption results, even if compromised. However, the relayers are still trusted to forward decryption requests on time.
+Relayers are the only accounts authorized to fulfill decryption requests. The role of the `DecryptionORacle`, however, is to independently verify the KMS signature during execution. This ensures that the relayers cannot manipulate or send fraudulent decryption results, even if compromised. However, the relayers are still trusted to forward decryption requests on time.
```

Comment:
It's not the case anymore. Anyone can fulfill the decryption requests if they provide correct signatures. And the callback is call directly. Before we were calling the Oracle which was calling the dapp. Now, anyone can call the dapp.